README.md

```diff
@@ -1,407 +1,414 @@
+| `<C-t>`        | go to a file in a new tab        |
+| `<C-u>`        | scroll up in preview window      |
+| `<C-d>`        | scroll down in preview window    |
+| `<C-c>`        | close telescope                  |
+| `<Esc>`        | close telescope (in normal mode) |
+
+To see the full list of mappings, check out `lua/telescope/mappings.lua` and
+the `default_mappings` table.  
+
+Below are examples of how you can change mappings globally.
 
 ```lua
 local actions = require('telescope.actions')
 
-require('telescope').setup {
+-- If you want your function to run after another action you should define it as follows
+local test_action = actions._transform_action(function(prompt_bufnr)
+  print("This function ran after another action. Prompt_bufnr: " .. prompt_bufnr)
+  -- Enter your function logic here. You can take inspiration from lua/telescope/actions.lua
+end)
+
+require('telescope').setup{
   defaults = {
     mappings = {
       i = {
-        -- Disable the default <c-x> mapping
+        -- To disable a keymap, put [map] = false
+        -- So, to not map "<C-n>", just put
         ["<c-x>"] = false,
-
-        -- Create a new <c-s> mapping
-        ["<c-s>"] = actions.goto_file_selection_split,
+        -- Otherwise, just set the mapping to the function that you want it to be.
+        ["<C-i>"] = actions.goto_file_selection_split,
+        -- Add up multiple actions
+        ["<CR>"] = actions.goto_file_selection_edit + actions.center,
+        -- You can perform as many actions in a row as you like
+        ["<CR>"] = actions.goto_file_selection_edit + actions.center + test_action,
+      },
+      n = {
+        ["<esc>"] = actions.close
       },
     },
   }
 }
-```
-
-To override only SOME of the default mappings, you can use the `attach_mappings` key in the `setup` table. For example:
-
-```lua
-function my_custom_picker(results)
-  pickers.new(opts, {
-    prompt_title = 'Custom Picker',
-    finder = finders.new_table(results),
-    sorter = sorters.fuzzy_with_index_bias(),
-    attach_mappings = function(_, map)
-      -- Map "<CR>" in insert mode to the funciton, actions.set_command_line
-      map('i', '<CR>', actions.set_command_line)
 
-      return true
-    end,
-  }):find()
+-- You can also define your own functions, which then can be mapped to a key
+local function test_action(prompt_bufnr)
+  print("Action was attached with prompt_bufnr: ", prompt_bufnr)
+-- Enter your function logic here. You can take inspiration from lua/telescope/actions.lua
 end
+["<C-i>"] = test_action,
+
+-- If you want your function to run after another action you should define it as follows
+local test_action = actions._transform_action(function(prompt_bufnr)
+  print("This function ran after another action. Prompt_bufnr: " .. prompt_bufnr)
+-- Enter your function logic here. You can take inspiration from lua/telescope/actions.lua
+end)
+["<C-i>"] = actions.goto_file_selection_split + test_action
+```
```

Comment:
This is our example for creating mappings. Lines 225-229 are identical to lines 259-264 (plus this mapping but we have this at Line 241). So i think we don't need the 259-264 one because it is pretty clear in the example above.

Lines 252-257 does the same as 225-229 but you can't add this action after another one. So something like `["<C-i>"] = actions.goto_file_selection_split + test_action` is not possible. I would suggest we just not even advertise this, so that users create actions always with `actions._transform_action` even though they don't use the overloaded `+` operator

Reply:
Oh I see where the duplication is. I've rewritten that part. Please check it.

For the other part , you mean that + only works with actions.___ and not custom functions??

Reply:
No it works with custom functions but only functions what are written like that:
```lua
local test_action = actions._transform_action(function(prompt_bufnr)
  -- Code
end)
```

not
```lua
local function test_action(prompt_bufnr)
  -- Code
end
```

That why i was asking if we can just advertise the first one which uses the `actions._transform_action` so that users always do that

Reply:
Yes @Conni2461 its better to use the first one. Thanks so much for this important notes.